src/components/tables/RowActions/Shared/ConfirmationWithExplination.tsx

```diff
@@ -0,0 +1,144 @@
+    useTheme,
+} from '@mui/material';
+import MessageWithLink from 'components/content/MessageWithLink';
+import AlertBox from 'components/shared/AlertBox';
+import { alertColorsReversed } from 'context/Theme';
+import { AccessGrantRemovalType } from 'hooks/useAccessGrantRemovalDescriptions';
+import { WarningTriangle } from 'iconoir-react';
+import { useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import { AccessGrantRowConfirmation } from '../AccessGrants/types';
+import { ConfirmationWithExplinationProps } from './types';
+
+function ConfirmationWithExplination({
+    message,
+    selected,
+}: ConfirmationWithExplinationProps) {
+    const intl = useIntl();
+    const theme = useTheme();
+
+    const { dangerous: potentiallyDangerousUpdates, normal: normalUpdates } =
+        useMemo(() => {
+            const response: {
+                [k in AccessGrantRemovalType]: AccessGrantRowConfirmation[];
+            } = {
+                dangerous: [],
+                normal: [],
+            };
+
+            selected.forEach((datum) => {
+                response[datum.details[0]].push(datum);
+            });
+
+            return response;
+        }, [selected]);
+
+    const renderTableRowContent = (item: AccessGrantRowConfirmation) => {
+        const dangerous = item.details[0] === 'dangerous';
+
+        const color = dangerous
+            ? alertColorsReversed.warning[theme.palette.mode]
+            : undefined;
+        const fontWeight = dangerous ? 500 : undefined;
+
+        return (
+            <>
+                <TableCell>{item.message}</TableCell>
+
+                <TableCell
+                    sx={{
+                        color,
+                        fontWeight,
+                        pr: 0,
+                        textAlign: 'right',
+                    }}
+                >
+                    {dangerous ? <WarningTriangle /> : null}
+                </TableCell>
+
+                <TableCell
+                    sx={{
+                        color,
+                        fontWeight,
+                    }}
+                >
+                    {item.details[1]}
+                </TableCell>
+            </>
+        );
+    };
+
+    return (
+        <Box>
+            {potentiallyDangerousUpdates.length > 0 ? (
+                <AlertBox
+                    short={false}
+                    sx={{
+                        my: 1,
+                        p: 3,
+                    }}
+                    headerMessage={intl.formatMessage({
+                        id: 'accessGrants.actions.extra.confirmation.title',
+                    })}
+                    severity="warning"
+                >
+                    <Typography component="div">
+                        <MessageWithLink messageID="accessGrants.actions.extra.confirmation.instructions" />
+                    </Typography>
+                </AlertBox>
+            ) : (
+                message
+            )}
```

Comment:
We want to show the huge alert if there are warnings to show. Otherwise, so the normal message. We might end up removing the fallback message.